src/paginationCalls/pageDescriptor.ts

```diff
@@ -14,158 +14,165 @@
-      apiResp: {}
-    ) {
-      if (err) {
-        stream.emit('error', err);
-        return;
-      }
-      // emit full api response with every page.
-      stream.emit('response', apiResp);
-      for (let i = 0; i < resources.length; ++i) {
-        if (ended(stream)) {
-          return;
-        }
-        if (resources[i] === null) {
-          continue;
-        }
-        stream.push(resources[i]);
-        pushCount++;
-        if (pushCount === maxResults) {
-          stream.end();
-        }
-      }
-      if (ended(stream)) {
-        return;
-      }
-      if (!next) {
-        stream.end();
-        return;
-      }
-      // When pageToken is specified in the original options, it will overwrite
-      // the page token field in the next request. Therefore it must be cleared.
-      if ('pageToken' in options) {
-        delete options.pageToken;
-      }
-      if (stream.isPaused()) {
-        request = next;
-        started = false;
-      } else {
-        setImmediate(apiCall, next, options, callback);
-      }
-    }
-    stream.on('resume', () => {
-      if (!started) {
-        started = true;
-        apiCall(request, options, callback as unknown as APICallback);
-      }
-    });
-    return stream;
-  }
-
-  /**
-   * Create an async iterable which can be recursively called for data on-demand.
-   */
-  asyncIterate(
-    apiCall: GaxCall,
-    request: RequestType,
-    options?: CallSettings
-  ): AsyncIterable<{} | undefined> {
-    options = Object.assign({}, options, {autoPaginate: false});
-    const iterable = this.createIterator(apiCall, request, options);
-    return iterable;
-  }
-
-  createIterator(
-    apiCall: GaxCall,
-    request: RequestType,
-    options: CallSettings
-  ): AsyncIterable<{} | undefined> {
-    const asyncIterable = {
-      [Symbol.asyncIterator]() {
-        let nextPageRequest: RequestType | null | undefined = request;
-        const cache: {}[] = [];
-        return {
-          async next() {
-            if (cache.length > 0) {
-              return Promise.resolve({
-                done: false,
-                value: cache.shift(),
-              });
-            }
-            if (nextPageRequest) {
-              let result: {} | [ResponseType] | null;
-              [result, nextPageRequest] = (await apiCall(
-                nextPageRequest!,
-                options
-              )) as ResultTuple;
-              cache.push(...(result as ResponseType[]));
-            }
-            if (cache.length === 0) {
-              return Promise.resolve({done: true, value: undefined});
-            }
-            return Promise.resolve({done: false, value: cache.shift()});
-          },
-        };
-      },
-    };
-    return asyncIterable;
-  }
-
-  getApiCaller(settings: CallSettings): APICaller {
-    if (!settings.autoPaginate) {
-      return new NormalApiCaller();
-    }
-    return new PagedApiCaller(this);
-  }
-}
+ import * as ended from 'is-stream-ended';
```

Comment:
We should not have any change in this file.

Reply:
You could run `git checkout origin/master -- path/to/this/file` to get master version of this file.

Reply:
call.ts, resourceCollector.ts, and pageDescriptor.ts were all reverted back to their master forms with git checkout origin/master. Any "changes" in the latest commit are my changes to call.ts and additional changes from that command.